airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/abstract_file_based_stream.py

```diff
@@ -0,0 +1,113 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from abc import abstractmethod
+from functools import cached_property
+from typing import Any, Dict, Iterable, List, Mapping, Optional, Union
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.file_based.availability_strategy import AbstractFileBasedAvailabilityStrategy
+from airbyte_cdk.sources.file_based.discovery_policy import AbstractDiscoveryPolicy
+from airbyte_cdk.sources.file_based.exceptions import ConfigValidationError, UndefinedParserError
+from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader
+from airbyte_cdk.sources.file_based.file_types.file_type_parser import FileTypeParser
+from airbyte_cdk.sources.file_based.remote_file import FileType, RemoteFile
+from airbyte_cdk.sources.file_based.stream.file_based_stream_config import FileBasedStreamConfig
+from airbyte_cdk.sources.streams import Stream
+
+
+class AbstractFileBasedStream(Stream):
+    """
+    A file-based stream in an Airbyte source.
+
+    In addition to the base Stream attributes, a file-based stream has
+    - A config object (derived from the corresponding stream section in source config).
+      This contains the globs defining the stream's files.
+    - A StreamReader, which knows how to list and open files in the stream.
+    - A FileBasedAvailabilityStrategy, which knows how to verify that we can list and open
+      files in the stream.
+    - A DiscoveryPolicy that controls the number of concurrent requests sent to the source
+      during discover, and the number of files used for schema discovery.
+    - A dictionary of FileType:Parser that holds all of the file types that can be handled
+      by the stream.
+    """
+
+    def __init__(
+        self,
+        raw_config: Dict[str, Any],
+        stream_reader: AbstractFileBasedStreamReader,
+        availability_strategy: AbstractFileBasedAvailabilityStrategy,
+        discovery_policy: AbstractDiscoveryPolicy,
+        parsers: Dict[FileType, FileTypeParser],
+    ):
+        super().__init__()
+        try:
+            self.config = FileBasedStreamConfig(**raw_config)
```

Comment:
why not ask the consumer to pass in a FileBasedStreamConfig?

Reply:
Sure, done.